spec/api/process_spec.rb

```diff
@@ -1528,6 +1528,78 @@
     end
   end
 
+  context "when retrieving a silenced registry entry by it's id" do
+    it "can retrieve entry for silencing a specific check on all clients" do
+      api_test do
+        options = { :body => { :check => "test" } }
+        api_request("/silenced", :post, options) do |http, body|
+          api_request("/silenced/ids/*:test") do |http, body|
+            expect(http.response_header.status).to eq(200)
+            expect(body).to be_kind_of(Hash)
+            expect(body[:id]).to eq("*:test")
+            async_done
+          end
+        end
+      end
+    end
+
+    it "can retrieve entry for silencing all checks on a specific subscription" do
+      api_test do
+        options = { :body => { :subscription => "test" } }
+        api_request("/silenced", :post, options) do |http, body|
+          api_request("/silenced/ids/test:*") do |http, body|
+            expect(http.response_header.status).to eq(200)
+            expect(body).to be_kind_of(Hash)
+            expect(body[:id]).to eq("test:*")
+            async_done
+          end
+        end
+      end
+    end
+
+    it "can retrieve entry for silencing all checks on all subscriptions" do
```

Comment:
Depending on perspective the bug might be that `:check => '*'` should be valid, and that `*:*`should silence everything, or that `:subscription => '*'` shouldn't be valid and silencing every check on every subscription is out of scope.